x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/HttpRequestSenderFactory.java

```diff
@@ -27,30 +36,73 @@
 public class HttpRequestSenderFactory {
     private final ThreadPool threadPool;
     private final HttpClientManager httpClientManager;
+    private final ClusterService clusterService;
+    private final Settings settings;
 
-    public HttpRequestSenderFactory(ThreadPool threadPool, HttpClientManager httpClientManager) {
+    public HttpRequestSenderFactory(
+        ThreadPool threadPool,
+        HttpClientManager httpClientManager,
+        ClusterService clusterService,
+        Settings settings
+    ) {
         this.threadPool = Objects.requireNonNull(threadPool);
         this.httpClientManager = Objects.requireNonNull(httpClientManager);
+        this.clusterService = Objects.requireNonNull(clusterService);
+        this.settings = Objects.requireNonNull(settings);
     }
 
     public HttpRequestSender createSender(String serviceName) {
-        return new HttpRequestSender(serviceName, threadPool, httpClientManager);
+        return new HttpRequestSender(serviceName, threadPool, httpClientManager, clusterService, settings);
     }
 
     /**
      * A class for providing a more friendly interface for sending an {@link HttpUriRequest}. This leverages the queuing logic for sending
      * a request.
      */
     public static final class HttpRequestSender implements Closeable {
+        private static final Logger logger = LogManager.getLogger(HttpRequestSender.class);
+
+        /**
+         * The maximum time a request can take. The timer starts once a request is enqueued and continues until a response is
```

Comment:
++ it makes sense to start the timer here

Reply:
The timer gets started when the `RequestTask` is constructed in the `HttpRequestExecutorService::start`.